Client/Ecosia/Extensions/HomepageViewController+Ecosia.swift

```diff
@@ -7,38 +7,24 @@ import Core
 
 protocol HomepageViewControllerDelegate: AnyObject {
     func homeDidTapSearchButton(_ home: HomepageViewController)
-    func homeDidPressPersonalCounter(_ home: HomepageViewController, completion: (() -> Void)?)
+    func showSettingsWithDeeplink(to destination: AppSettingsDeeplinkOption)
 }
 
-extension HomepageViewController {
-    func configureEcosiaSetup() {
-        personalCounter.subscribe(self) { [weak self] _ in
-            guard let self = self else { return }
-            self.updateTreesCell()
-        }
-
-        referrals.subscribe(self) { [weak self] _ in
-            guard let self = self else { return }
-            self.updateTreesCell()
-        }
-    }
-
-    func updateTreesCell() {
-        guard let impactCell = viewModel.impactViewModel.cell else { return }
-        impactCell.display(treesCellModel, animated: false)
-    }
+protocol SharedHomepageCellDelegate: AnyObject {
+    func openLink(url: URL)
+    func invalidateLayout(at indexPaths: [IndexPath])
+}
 
-    var treesCellModel: NTPImpactCell.Model {
-        .init(impact: User.shared.impact, searches: personalCounter.state!, trees: TreeCounter.shared.treesAt(.init()))
+extension HomepageViewController: SharedHomepageCellDelegate {
+    func openLink(url: URL) {
+        homePanelDelegate?.homePanel(didSelectURL: url, visitType: .link, isGoogleTopSite: false)
```

Comment:
Totally agree that the signature could be better 👍 Even so `HomePanelDelegate` protocol is actually Firefox code, so I would vote against renaming them if not strictly needed to prevent conflicts in future updates. What do you think in that case?

Reply:
Same as above ☝️ - Let's leave it as it then 💪 - it's ok as perhaps FF codebase updated it